core/src/main/java/com/oath/oak/OakMap.java

```diff
@@ -98,95 +98,62 @@ private OakMap(InternalOakMap internalOakMap, MemoryManager memoryManager,
-    public long memorySize() {
-        return internalOakMap.memorySize();
-    }
+    /* ------ Map API methods ------ */
 
     /**
-     * @return the current number of keys in the map.
-     * Isn't supported for SubMaps
+     * {@inheritDoc}
+     * @throws UnsupportedOperationException if used on a SubMap
      */
-    public int entries() {
-        if (this.fromKey != null || this.toKey != null) {
-            // this is a SubMap, for SubMap number of keys can not be counted
-            throw new UnsupportedOperationException();
-        }
+    @Override
+    public int size() {
+        if (this.isSubmap()) throw new UnsupportedOperationException();
 
         return internalOakMap.entries();
     }
 
-    /* ------ Map API methods ------ */
-
     /**
-     * Associates the specified value with the specified key in this map.
-     * If the map previously contained a mapping for the key, the old
-     * value is replaced.
-     * Creates a copy of the value in the map.
+     * Returns a read only view of the value to which the specified key is mapped,
+     * or {@code null} if this map contains no mapping for the key.
      *
-     * @param key   key with which the specified value is to be associated
-     * @param value value to be associated with the specified key
+     * @param key the key whose associated value is to be returned
+     * @return the value associated with that key, or
+     * {@code null} if this map contains no mapping for the key
      * @throws IllegalArgumentException if the specified key is null
      */
-    public void put(K key, V value) {
-        if (key == null || value == null)
+    @Override
+    public V get(Object key) {
+        // (eranmeir) API note: this assume the map does not contains null values.
+        // This assumption should be documented. If it is changed, getOrDefault should be overridden
+        if (key == null)
             throw new IllegalArgumentException();
         if (!inBounds(key))
             throw new IllegalArgumentException("The key is out of map bounds");
-        internalOakMap.put(key, value);
+
+        return (V) internalOakMap.getValueTransformation(key, valueDeserializeTransformer);
     }
 
-    /**
-     * If the specified key is not already associated
-     * with a value, associate it with the given value.
-     * Creates a copy of the value in the map.
-     *
-     * @param key   key with which the specified value is to be associated
-     * @param value value to be associated with the specified key
-     * @return {@code true} if there was no mapping for the key
-     * @throws IllegalArgumentException if the specified key or value is null
-     */
-    public boolean putIfAbsent(K key, V value) {
-        if (key == null || value == null)
-            throw new IllegalArgumentException();
-        if (!inBounds(key))
-            throw new IllegalArgumentException("The key is out of map bounds");
-        return internalOakMap.putIfAbsent(key, value);
```

Comment:
There are checks in internal oak map, but I also added checks in the API wrappers themselves, since some requirements may be different (e.g, checking if a transformer is null).